src/UsersModule/Components/LandingPage/LandingPage.tsx

```diff
@@ -1,17 +1,941 @@
+import hotel3 from "../../../assets/hotel3.png";
+import hotel4 from "../../../assets/hotel4.png";
+
+import house1 from "../../../assets/house1.png";
+import house2 from "../../../assets/house2.png";
+import house3 from "../../../assets/house3.png";
+import house4 from "../../../assets/house4.png";
+
+import landingImg from "../../../assets/landingImg.png";
+import { useForm } from "react-hook-form";
+import happyFamily from "../../../assets/happyFamily.png";
+import AnonymousUserAlert from "../AnonymousUserAlert/AnonymousUserAlert";
+import { toast } from "react-toastify";
+import NoData from "../../../SharedModule/Components/NoData/NoData";
+import Loading from "../../../SharedModule/Components/Loading/Loading";
 
 export default function LandingPage() {
-  const navigate=useNavigate();
+  const navigate = useNavigate();
+  const [ads, setAds] = React.useState([]);
+  let fiveAds = ads?.slice(0, 5);
+  const [isLoading, setIsLoading] = useState(false);
+
+
+  const [anonymousAlert, setAnonymousAlert] = useState(false);
+  const handleCloseAnonymousAlert = () => setAnonymousAlert(false);
+  const handleOpenAnonymousAlert = () => setAnonymousAlert(true);
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm();
+
+  async function getAds() {
+    setIsLoading(true);
+    try {
+      const response = await axios.get(`${baseUrl}/portal/ads`);
+      console.log(response?.data?.data?.ads);
+      setAds(response?.data?.data?.ads);
+    } catch (error) {
+      console.log(error);
+    }
+    setIsLoading(false);
+  }
+
+  type ExploreData = {
+    startDate: Date;
+    endDate: Date;
+    capacity: number;
+  };
+
+  const Explore = {
+    startDate: new Date(),
+    endDate: new Date(),
+    capacity: new Number(),
+  };
+
+  async function handleExplore(data: ExploreData) {
+    // console.log("explore ", data);
+    Explore.startDate = data.startDate;
+    Explore.endDate = data.endDate;
+    Explore.capacity = data.capacity;
+    // console.log(Explore);
+    navigate(`all-rooms/${JSON.stringify(Explore)}`);
+  }
+
+  async function addtofav(roomId: string) {
+    let token = localStorage.getItem("adminToken");
+    let role: string = JSON.parse(localStorage.getItem("userRole"));
+
+    if (role == "user") {
+      try {
+        const response = await axios.post(
+          `${baseUrl}/portal/favorite-rooms`,
+          { roomId: roomId },
+          { headers: { Authorization: token } }
+        );
+        // console.log(response);
+        toast.success("Room Added To Your Favourits Successfuly");
+      } catch (error) {
+        // console.log(error);
+        toast.error(error?.response?.data?.message||'Somthing Went Wrong!');
+      }
+    } else {
+      handleOpenAnonymousAlert();
+    }
+  }
+
+  function goToRoomDetails(roomId: string) {
+
+      navigate(`all-rooms/room-details/${JSON.stringify(roomId)}`);
+  }
+
+  useEffect(() => {
+    
+    getAds();
```

Comment:
When using variables or functions in a useEffect, we should include them in the dependency array to avoid any potential bugs, but since we only use `getAds` in the `useEffect` we can declare the function in the useEffect  instead.